packages/lodestar/src/network/gossip/validation/attestation.ts

```diff
@@ -31,73 +32,86 @@ export async function validateGossipAttestation(
   logger.verbose("Started gossip committee attestation validation", attestationLogContext);
 
   if (!isUnaggregatedAttestation(attestation)) {
-    logger.warn("Rejected gossip committee attestation", {
-      reason: "not unaggregated attesation",
-      aggregationBits: JSON.stringify(attestation.aggregationBits),
+    throw new AttestationError({
+      code: AttestationErrorCode.ERR_NOT_EXACTLY_ONE_AGGREGATION_BIT_SET,
+      aggregationBits: attestation.aggregationBits,
       ...attestationLogContext,
+      job: attestationJob,
     });
-    return ExtendedValidatorResult.reject;
   }
-
   if (await isAttestingToInValidBlock(db, attestation)) {
-    logger.warn("Rejected gossip committee attestation", {
-      reason: "attestation block is invalid",
+    throw new AttestationError({
+      code: AttestationErrorCode.ERR_KNOWN_BAD_BLOCK,
       ...attestationLogContext,
+      job: attestationJob,
     });
-    return ExtendedValidatorResult.reject;
   }
 
   if (!hasValidAttestationSlot(config, chain.clock.currentSlot, attestation.data.slot)) {
-    logger.warn("Ignored gossip committee attestation", {reason: "Invalid slot time", ...attestationLogContext});
     // attestation might be valid later so passing to attestation pool
     await chain.receiveAttestation(attestation);
-    return ExtendedValidatorResult.ignore;
+
+    throw new AttestationError({
+      code: AttestationErrorCode.ERR_SLOT_OUT_OF_RANGE,
+      ...attestationLogContext,
+      currentSlot: chain.clock.currentSlot,
+      job: attestationJob,
+    });
   }
 
   // no other validator attestation for same target epoch has been seen
   if (await db.seenAttestationCache.hasCommitteeAttestation(attestation)) {
-    return ExtendedValidatorResult.ignore;
+    throw new AttestationError({
+      code: AttestationErrorCode.ERR_ATTESTATION_ALREADY_KNOWN,
+      ...attestationLogContext,
+      root: attestation.data.beaconBlockRoot as Uint8Array,
+      job: attestationJob,
+    });
   }
 
   if (await db.badBlock.has(attestation.data.beaconBlockRoot.valueOf() as Uint8Array)) {
-    logger.warn("Rejecting gossip committee attestation", {
-      reason: "attestation attests known bad block",
+    throw new AttestationError({
+      code: AttestationErrorCode.ERR_KNOWN_BAD_BLOCK,
       ...attestationLogContext,
+      job: attestationJob,
     });
-    return ExtendedValidatorResult.reject;
   }
 
   if (!chain.forkChoice.hasBlock(attestation.data.beaconBlockRoot)) {
-    logger.warn("Ignored gossip committee attestation", {
-      reason: "missing attestation beaconBlockRoot block",
-      ...attestationLogContext,
-    });
     // attestation might be valid after we receive block
     await chain.receiveAttestation(attestation);
-    return ExtendedValidatorResult.ignore;
+
+    throw new AttestationError({
+      code: AttestationErrorCode.ERR_UNKNOWN_HEAD_BLOCK,
```

Comment:
this error has wierd name, we are missing attestation beacon block root, not head block